File: 2019/day01/part_2.rb
def adder(i, sum)
  n = (i/ 3) - 2
  return n <= 0 ? sum : adder(n, sum + n)
end

puts File.readlines("./input.txt")
        .map(&:strip)
        .map(&:to_i)
        .map { |x| adder(x, 0) }
        .reduce(0, :+)

File: 2019/day01/part_2.scala
import scala.tools.nsc.io.File
import scala.util.Try

object MyApp extends App {

  def add(i: Int, sum: Int): Int = i match {
    case x if x <= 0 =>
      sum
    case x =>
      val n = x/3 - 2
      add(n, sum + n)
  }


  val f = File("./input.txt")
            .lines
            .map(_.trim)
            .map(x => Try(add(x.toInt, 0)).getOrElse(0))
            .sum

  println(s"Sum is $f")
}

